src/signer/canister/tests/it/utils/pic_canister.rs

```diff
@@ -0,0 +1,279 @@
+            })
+    }
+
+    /// Makes a query call to the canister.
+    #[allow(dead_code)]
+    fn query<T>(&self, caller: Principal, method: &str, arg: impl CandidType) -> Result<T, String>
+    where
+        T: for<'a> Deserialize<'a> + CandidType,
+    {
+        self.pic()
+            .query_call(self.canister_id(), caller, method, encode_one(arg).unwrap())
+            .map_err(|e| {
+                format!(
+                    "Query call error. RejectionCode: {:?}, Error: {}",
+                    e.code, e.description
+                )
+            })
+            .and_then(|reply| match reply {
+                WasmResult::Reply(reply) => {
+                    decode_one(&reply).map_err(|_| "Decoding failed".to_string())
+                }
+                WasmResult::Reject(error) => Err(error),
+            })
+    }
+    fn workspace_dir() -> PathBuf {
+        let output = std::process::Command::new(env!("CARGO"))
+            .arg("locate-project")
+            .arg("--workspace")
+            .arg("--message-format=plain")
+            .output()
+            .unwrap()
+            .stdout;
+        let cargo_path = Path::new(std::str::from_utf8(&output).unwrap().trim());
+        cargo_path.parent().unwrap().to_path_buf()
+    }
+    /// The path to a typical Cargo Wasm build.
+    #[allow(dead_code)]
+    fn cargo_wasm_path(name: &str) -> String {
+        let workspace_dir = Self::workspace_dir();
+        workspace_dir
+            .join("target/wasm32-unknown-unknown/release")
+            .join(name)
+            .with_extension("wasm")
+            .to_str()
+            .unwrap()
+            .to_string()
+    }
+    /// The path to the wasm after `dfx deploy`.  Expects the Wasm to be gzipped.
+    ///
+    /// If not already gzipped, please add this to the canister declaration in `dfx.json`: `"gzip": true`
+    #[allow(dead_code)]
+    fn dfx_wasm_path(name: &str) -> String {
+        Self::workspace_dir()
+            .join(format!(".dfx/local/canisters/{name}/{name}.wasm.gz"))
+            .to_str()
+            .unwrap()
+            .to_string()
+    }
+}
+
+/// A typical canister running on PocketIC.
+pub struct PicCanister {
+    pub pic: Arc<PocketIc>,
+    pub canister_id: Principal,
+}
+
+impl PicCanisterTrait for PicCanister {
+    /// The shared PocketIc instance.
+    fn pic(&self) -> Arc<PocketIc> {
+        self.pic.clone()
+    }
+    /// The ID of this canister.
+    fn canister_id(&self) -> Principal {
+        self.canister_id.clone()
+    }
+}
+
+impl PicCanister {
+    /// Creates a new canister.
+    pub fn new(pic: Arc<PocketIc>, wasm_path: &str) -> Self {
+        PicCanisterBuilder::default()
+            .with_wasm(wasm_path)
+            .deploy_to(pic)
+    }
+}
+
+/// Canister installer, using the builder pattern, for use in test environmens using `PocketIC`.
+///
+/// # Example
+/// For a default test environment:
+/// ```
+/// let pic_canister = BackendBuilder::default().deploy();
```

Comment:
Do you mean `PicCanisterBuilder` instead of `BackendBuilder`?

Reply:
What is `pic_canister`? Isn't it a tuple `(pic, canister_id)`?

Reply:
Yes, it should be PicCanisterBuilder.  I already had the change locally but the push failed, so good catch.

Reply:
The tuples need to be updated.  I'll go through the rustdoc and get it updated.

Reply:
Done.  And added some tests for deployment to pocket ic.